core/schema/service.go

```diff
@@ -105,14 +123,94 @@ func (s *Service) Create(ctx context.Context, nsName string, schemaName string,
 		Compatibility: compatibility,
 	}
 	versionID := getIDforSchema(nsName, schemaName, sf.ID)
+	_, prevSchemaData, dataErr := s.GetLatest(ctx, nsName, schemaName)
 	version, err := s.repo.Create(ctx, nsName, schemaName, mergedMetadata, versionID, sf)
+	if dataErr == nil {
+		changeRequest := &changedetector.ChangeRequest{
+			NamespaceID: nsName,
+			SchemaName:  schemaName,
+			Version:     version,
+			VersionID:   versionID,
+			OldData:     prevSchemaData,
+			NewData:     data,
+			Depth:       s.config.SchemaChange.Depth,
+		}
+		go func() {
+			newCtx, cancel := context.WithTimeout(context.Background(), 1*time.Hour)
+			defer cancel()
+			err := s.identifySchemaChange(newCtx, changeRequest)
+			if err != nil {
+				log.Printf("got error while identifying schema change event %s", err.Error())
+			}
+		}()
+	} else {
+		log.Printf("got error while getting previous schema data %s", dataErr.Error())
+	}
 	return SchemaInfo{
 		Version:  version,
 		ID:       versionID,
 		Location: fmt.Sprintf("/v1beta1/namespaces/%s/schemas/%s/versions/%d", nsName, schemaName, version),
 	}, err
 }
 
+func (s *Service) identifySchemaChange(ctx context.Context, request *changedetector.ChangeRequest) error {
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	default:
+		endFunc := s.newrelic.StartGenericSegment(ctx, "Identify Schema Change")
```

Comment:
Instead of moving whole implementation under `default` can we have one more function which handles context related and call this function.

This function would remain like it was before.

Reply:
done